DevoxxClientMobile/src/main/java/com/devoxx/util/DevoxxNotifications.java

```diff
@@ -145,9 +240,12 @@ public void preloadFavoriteSessions() {
                 }
             };
             service.retrieveFavoredSessions().addListener(favoriteSessionsListener);
+            // we don't make call to preloadRatingNotifications from DevoxxService#retrieveSessionsInternal
+            // when the user is authenticated and try to fuse the flow with that of FavoriteNotifications
+            preloadRatingNotifications();
```

Comment:
The only problem I see now is if the call to `preloadingFavoriteSessionsDone` happens before `preloadRatingNotifications` ends (for instance, if case there are no favored sessions), you will miss the rating notification from the map

Reply:
This is the case where favorites are retrieved before sessions is retrieved, right?

Reply:
Yes, you have an authenticated user, and favored sessions is empty, for instance. Then `preloadingFavoriteSessionsDone` can be called way sooner than `preloadRatingNotifications` ends adding the rating notification to the map, so it won't be available to the notifications map

Reply:
Do you have a suggestion on how to fix this? I can add flags to DevoxxService, so that `preloadingFavoriteSessionsDone` is only called after both favorites and sessions are done loading but it would be very ugly.

Reply:
Not really, unless you want to do one first and when done do the other.
 For instance, any type of user will have a rating notification, so that goes first (although retrieving the full list of sessions might take more time), and when done, if authenticated start doing the favored sessions flow, else call ratingDone.

Reply:
I merged both the `*done()` and since it doesn't require authentication. Check this [commit](https://github.com/abhinayagarwal/MyDevoxxGluon/pull/2/commits/fd02382b2bb944775eda1c618c86770d9a0dc2bc)

In Service, I call `finishNotificationsPreloading` method from both fav and session. The method calls `DevoxxNotifications#preloadingNotificationsDone` only when the both are done loading the sessions from the RF.

Reply:
I've tested it with some favored and rating notifications.
I haven't done an extensive test, but I had 1 rating notification, 1 start session notification and 1 vote notification. I've opened and closed the app a few times in between.
I've received the three notifications always with the app closed. The three of them opened the app, and:
+ the rating notification showed the rating control on top of the session catalog.
- The start session failed to go to the session
+ The vote session went to the vote pane. 

I believe the failure was related to `finishNotificationsPreloading` not being called, or if called the if condition wasn't true.

I can keep testing, and check if the failure happens again.

Reply:
Let me know when you think its safe to merge this PR and update the PR in the parent repo :)